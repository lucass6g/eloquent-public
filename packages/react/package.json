{
  "name": "@eloquent/react",
  "version": "0.0.1",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.0.2",
    "@types/react": "^18.2.66",
    "@types/react-dom": "^18.2.22",
    "react": "^18.2.0",
    "typescript": "^5.4.2"
  }
}
