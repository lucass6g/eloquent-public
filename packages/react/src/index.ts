export {Switch, Case, Default} from './components/conditional-switch'