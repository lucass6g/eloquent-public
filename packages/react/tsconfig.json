{
  "extends": "../../packages/config/base.tsconfig.json",
  "include": ["src"],
  "exclude": ["dist", "build", "node_modules"],
  "compilerOptions": {
    "forceConsistentCasingInFileNames": false,
    "rootDir": "src",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
