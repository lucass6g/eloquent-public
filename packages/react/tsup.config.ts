import { defineConfig } from 'tsup';

export default defineConfig({
	name: "@eloquent/react",
	entry: ["src/index.ts"],
	outDir: "dist",
	target: "es2022",
	format: ["cjs", "esm"],
	splitting: false,
	sourcemap: true,
	clean: true,
	dts: true,
	external: ["react", 'react-dom', 'react/jsx-runtime'],
	cjsInterop: true,
	treeshake: true,
})