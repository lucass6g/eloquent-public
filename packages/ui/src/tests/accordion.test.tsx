import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import { Accordion } from "..";

describe("Accordion", () => {
    it("should match the snapshot", () => {
        const component = render(<Accordion.Root type="single" collapsible>
            <Accordion.Item value="item-1">
                <Accordion.Trigger>teste</Accordion.Trigger>
                <Accordion.Content>teste</Accordion.Content>
            </Accordion.Item>
        </Accordion.Root>);

        expect(component.baseElement).toMatchSnapshot()
    })
}); 