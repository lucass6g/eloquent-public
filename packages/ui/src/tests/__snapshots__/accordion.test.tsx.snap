// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Accordion > should match the snapshot 1`] = `
<body>
  <div>
    <div
      data-orientation="vertical"
    >
      <div
        class="border-b"
        data-orientation="vertical"
        data-state="closed"
      >
        <h3
          class="flex"
          data-orientation="vertical"
          data-state="closed"
        >
          <button
            aria-controls="radix-:r1:"
            aria-expanded="false"
            class="flex flex-1 items-center justify-between py-4 text-green-dark-600 text-sm font-medium transition-all hover:underline [&[data-state=open]>svg]:rotate-180"
            data-orientation="vertical"
            data-radix-collection-item=""
            data-state="closed"
            id="radix-:r0:"
            type="button"
          >
            teste
            <svg
              class="h-4 w-4 shrink-0 text-muted-foreground transition-transform duration-200"
              fill="none"
              height="15"
              viewBox="0 0 15 15"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </h3>
        <div
          aria-labelledby="radix-:r0:"
          class="overflow-hidden text-sm data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down"
          data-orientation="vertical"
          data-state="closed"
          hidden=""
          id="radix-:r1:"
          role="region"
          style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
        />
      </div>
    </div>
  </div>
</body>
`;
